pkg/agent/flowexporter/exporter/exporter.go

```diff
@@ -175,23 +178,31 @@ func (exp *flowExporter) initFlowExporter(collectorAddr string, collectorProto s
 		return fmt.Errorf("cannot generate obsID for IPFIX ipfixexport: %v", err)
 	}
 
-	if strings.Contains(collectorAddr, flowAggregatorDNSName) {
-		hostIPs, err := net.LookupIP(flowAggregatorDNSName)
+	var expInput exporter.ExporterInput
+	if exp.enableTLSToFlowAggregator {
+		// if CA certificate, client certificate and key do not exist during initialization,
+		// it will retry to obtain the credentials in next export cycle
+		caCert, err := getCACert(exp.k8sClient)
 		if err != nil {
-			return err
+			return fmt.Errorf("cannot retrieve CA cert: %v", err)
 		}
-		// Currently, supporting only IPv4 for Flow Aggregator.
-		ip := hostIPs[0].To4()
-		if ip != nil {
-			// Update the collector address with resolved IP of flow aggregator
-			collectorAddr = net.JoinHostPort(ip.String(), defaultIPFIXPort)
-		} else {
-			return fmt.Errorf("resolved Flow Aggregator address %v is not supported", hostIPs[0])
+		clientCert, clientKey, err := getClientCertKey(exp.k8sClient)
+		if err != nil {
+			return fmt.Errorf("cannot retrieve client cert and key: %v", err)
 		}
-	}
-
-	var expInput exporter.ExporterInput
-	if collectorProto == "tcp" {
+		// TLS transport does not need any tempRefTimeout, so sending 0.
+		expInput = exporter.ExporterInput{
+			CollectorAddress:    flowAggregatorDNSName + ":4739",
```

Comment:
@zyiou BTW this should be `CollectorAddress : collectorAddr`. `flowAggregatorDNSName` should not be used anywhere in the Agent except as the default value for the collector address.

Reply:
Thanks for reminding. Updated. Also add a TODO in flowaggregator.go to support IP address in server certificate in the future.